js/modules/GLPI.js

```diff
@@ -0,0 +1,209 @@
+ * the Free Software Foundation, either version 3 of the License, or
+ * (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program.  If not, see <https://www.gnu.org/licenses/>.
+ *
+ * ---------------------------------------------------------------------
+ */
+
+import Clipboard from './Clipboard.js';
+import Dialogs from "./Dialogs.js";
+
+/**
+ * The core GLPI module code.
+ * This code should be loaded globally for all pages.
+ */
+
+// Save any previous value of GLPI.
+if (window.GLPI !== undefined) {
+    window._GLPI = window.GLPI;
+}
+// Create a window variable to loading modules from legacy scripts
+
+window.GLPILegacy = {
+    /** @type {{key: string, load_callback: function}[]} */
+    _load_queue: [],
+    loadModule(key, load_callback = undefined) {
+        if (load_callback === undefined) {
+            load_callback = () => {};
+        }
+        if (window.GLPI.registerKnownModule !== undefined) {
+            window.GLPI.registerKnownModule(key).then(load_callback);
+        } else {
+            this._load_queue.push({
+                key: key,
+                load_callback: load_callback
+            });
+        }
+    }
+};
+window.GLPI = new class GLPI {
+    constructor() {
+        /** @type {Object.<string, GLPIModulePlugin>} */
+        this.modules = {};
+
+        this.known_modules = {
+            'clipboard': new Clipboard(),
+            'dialogs': new Dialogs(),
+            'rack': 'DCIM/Rack',
+        };
+
+        /** @type {EventTarget} */
+        this.event_target = new EventTarget();
+
+        // Try loading all modules from legacy scripts
+        if (window.GLPILegacy._load_queue.length > 0) {
+            window.GLPILegacy._load_queue.forEach((m) => {
+                this.registerKnownModule(m.key).then(m.load_callback);
+            });
+        }
+    }
+
+    async initialize() {
+        await this.registerCoreModules();
+    }
+
+    async registerCoreModules() {
+        await Promise.all([
+            this.registerKnownModule('clipboard'),
+            this.registerKnownModule('dialogs'),
+        ]);
+    }
+
+    async registerKnownModule(key) {
+        if (this.known_modules[key] !== undefined) {
+            let m = this.known_modules[key];
+            if (typeof m === 'string') {
+                // Prevent directory traversal and other issues (Remove all characters that are not a-zA-Z0-9_- or forward slash)
+                m = m.replace(/[^a-zA-Z\d_\-/]/g, '');
+                // Load the module from a script
+                const { default: m_class } = await import('./' + m + '.js');
```

Comment:
In existing logic, a `v` parameter is used to prevent browser/proxy caching issues.
To prevent similar issues here, we should probably use this cache buster parameter on ES imports.
I found this solution (not tested):
```
const version = Math.random();
const url = CFG_GLPI.root_doc + '/js/modules/' + m + '.js?v=' + version;
const blob_url = URL.createObjectURL(await (await fetch(url)).blob());
const { default: m_class } = await import(blob_url);
```

`version` should be replaced by corresponding version of GLPI (or version of target plugin). It should be obfuscated using `Glpi\Toolbox\FrontEnd::getVersionCacheKey()` and probably put in `CFG_GLPI` javascript variable.